test/extended/builds/volumes.go

```diff
@@ -0,0 +1,203 @@
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By(fmt.Sprintf("updating build config %q script and build volumes", buildConfigName))
+				bc.Spec.PostCommit.Script = script
+				switch bc.Spec.Strategy.Type {
+				case buildv1.SourceBuildStrategyType:
+					bc.Spec.Strategy.SourceStrategy.Volumes = expectedBuildVolumes
+				case buildv1.DockerBuildStrategyType:
+					bc.Spec.Strategy.DockerStrategy.Volumes = expectedBuildVolumes
+				}
+				bc, err = oc.BuildClient().BuildV1().BuildConfigs(oc.Namespace()).Update(context.Background(), bc, metav1.UpdateOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By(fmt.Sprintf("ensuring that build config %q contains the expected build volumes", buildConfigName))
+				var bcBuildVolumes []buildv1.BuildVolume
+				switch bc.Spec.Strategy.Type {
+				case buildv1.SourceBuildStrategyType:
+					bcBuildVolumes = bc.Spec.Strategy.SourceStrategy.Volumes
+				case buildv1.DockerBuildStrategyType:
+					bcBuildVolumes = bc.Spec.Strategy.DockerStrategy.Volumes
+				}
+				o.Expect(reflect.DeepEqual(expectedBuildVolumes, bcBuildVolumes)).To(o.BeTrue())
+
+				g.By(fmt.Sprintf("starting a build from build config %q", buildConfigName))
+				br, _ := exutil.StartBuildAndWait(oc, buildConfigName)
+				br.AssertSuccess()
+
+				g.By(fmt.Sprintf("ensuring that the build volumes from build config %q are propogated to build %q", buildConfigName, br.BuildName))
+				var buildBuildVolumes []buildv1.BuildVolume
+				switch br.Build.Spec.Strategy.Type {
+				case buildv1.SourceBuildStrategyType:
+					buildBuildVolumes = br.Build.Spec.Strategy.SourceStrategy.Volumes
+				case buildv1.DockerBuildStrategyType:
+					buildBuildVolumes = br.Build.Spec.Strategy.DockerStrategy.Volumes
+				}
+				o.Expect(reflect.DeepEqual(expectedBuildVolumes, buildBuildVolumes)).To(o.BeTrue())
+
+				g.By(fmt.Sprintf("getting the pod for build %q", br.BuildName))
+				pod, err := oc.KubeClient().CoreV1().Pods(oc.Namespace()).Get(context.Background(), br.BuildName+"-build", metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By(fmt.Sprintf("ensuring that the build volumes from build %q are propogated to the build pod spec", br.BuildName))
+				for _, volume := range expectedVolumes {
+					found := false
+					for _, v := range pod.Spec.Volumes {
+						if reflect.DeepEqual(volume, v) {
+							found = true
+							break
+						}
+					}
+					o.Expect(found).To(o.BeTrue())
+				}
+
+				g.By(fmt.Sprintf("ensuring that the build volumes mounts from build %q are propogated to the build pod spec", br.BuildName))
+				for _, volumeMount := range expectedVolumeMounts {
+					found := false
+					for _, vm := range pod.Spec.Containers[0].VolumeMounts {
+						if reflect.DeepEqual(volumeMount, vm) {
+							found = true
+							break
+						}
+					}
+					o.Expect(found).To(o.BeTrue())
+				}
```

Comment:
This will make the e2e test brittle over time. From an end user perspective we really don't care how the volume gets to the build pod and then buildah's runtime. What users do care about is:

1. The `Secret` or `ConfigMap` data is readable during the build
2. The `Secret` or `ConfigMap` data is not present in the output image.

We have item 1 in this test, but not 2. I recommend that to verify this feature, we start a pod with the build's output image that tries to `cat` the contents of the data in the Secret and ConfigMap.

Reply:
updated in current push for s2i, will copy to docker test if it passes